stock_picking_limit_return_qty/wizard/stock_picking_return.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2023 Cetmix OU
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ReturnPickingLine(models.TransientModel):
+    _inherit = "stock.return.picking.line"
+
+    quantity_max = fields.Float(
+        "Maximum Quantity",
+        digits="Product Unit of Measure",
+        help="Maximum quantity that can be returned",
+    )
+
+    @api.constrains("quantity")
+    def _constraints_quantity(self):
+        for rec in self:
+            if rec.quantity and rec.quantity_max and rec.quantity > rec.quantity_max:
+                raise ValidationError(
+                    _(
+                        "You can return only %(max_qty)s of %(product)s",
+                        max_qty=rec.quantity_max,
+                        product=rec.product_id.name,
+                    )
+                )
+
+
+class ReturnPicking(models.TransientModel):
+    _inherit = "stock.return.picking"
+
+    @api.model
+    def _prepare_stock_return_picking_line_vals_from_move(self, stock_move):
+
+        res = super(
+            ReturnPicking, self
+        )._prepare_stock_return_picking_line_vals_from_move(stock_move)
+
+        # Store maximum quantity that is possible to return
+        res.update(
```

Comment:
As this is breaking the default Odoo's behaviour, I would have added a configuration parameter to activate this.

Either a global one, either one on stock.picking.type object (I don't know if it is defined on stock move values at this point)

Reply:
@rousseldenis thank you for mentioning this. Sounds reasonable. The only thing we didn't add it is because this is actually the only feature this module is adding. So this setting is actually if the module installed or not. Don't want this behaviour? Just uninstall it.
However the concern was that this module will be installed on the runboat bu default. So it might cause some confusion.  It could be a good idea if this module could be installed as using checkbox in settings like this is done with some core Odoo modules. However I didn't manage to find to do the same for the third party modules.
Would be great if you could give some hints.